packages/windows_application_windows.go

```diff
@@ -0,0 +1,122 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package packages
+
+import (
+	//"golang.org/x/sys/windows"
+	//"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+	//"time"
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/GoogleCloudPlatform/osconfig/clog"
+	"golang.org/x/sys/windows/registry"
+)
+
+func getStringValueReturnEmptyIfError(k *registry.Key, value string) string {
+	value, _, err := k.GetStringValue(value)
+	if err != nil {
+		return ""
+	}
+	return value
+}
+
+func parseDate(dateString string) time.Time {
+	var t time.Time
+	if len(dateString) != 8 {
+		return t
+	}
+
+	year, err := strconv.ParseInt(dateString[0:4], 10, 32)
+	if err != nil {
+		return t
+	}
+	month, err := strconv.ParseInt(dateString[4:6], 10, 32)
+	if err != nil {
+		return t
+	}
+	day, err := strconv.ParseInt(dateString[6:8], 10, 32)
+	if err != nil {
+		return t
+	}
+
+	return time.Date(int(year), time.Month(month), int(day), 0, 0, 0, 0, time.Now().Location())
+}
+
+func getWindowsApplicationFromRegistryKey(ctx context.Context, k *registry.Key) *WindowsApplication {
+	displayName, _, errName := k.GetStringValue("DisplayName")
+	_, _, errUninstall := k.GetStringValue("UninstallString")
```

Comment:
Do you have examples of entries that don't have "UninstallString"? I just want some good data on why we would not want to include these.

Reply:
I've been examining windows registry in these locations on my local machine (by running some powershell commands and examining the output), and can't find anything that's weird and suspicious. I'm not sure if it's good enough of a test, but based on that we can probably include these entries.

Reply:
My point is you check errUninstall == nil, it can see an entry not having "UninstallString" (probably not uncommon) and we would not record those applications

Reply:
Yep. I haven't found example of application that doesn't have UninstallString, but has other fields that actually make sense.

At the moment I'm seeing entries like: 

HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\MPlayer2
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\IE40
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\SchedulingAgent

that only have one entry:

| Name      | Type   | Data            |
|-----------|--------|-----------------|
| (Default) | REG_SZ | (value not set) |

This entry for example:
- HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Connection Manager

has:

| Name            | Type      | Data            |
|-----------------|-----------|-----------------|
| (Default)       | REG_SZ    | (value not set) |
| SystemComponent | REG_DWORD | 0x00000001 (1)  |

While this:
- HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\WIC

has:

| Name      | Type      | Data            |
|-----------|-----------|-----------------|
| (Default) | REG_SZ    | (value not set) |
| NoRemove  | REG_DWORD | 0x00000001 (1)  |

There are few more examples. Some of these can be considered "applications" like mplayer (I can actually start it and run it from windows start menu). Not quite sure what are the others but it seems like it's something installed by default on Windows (not using typical windows installer).